### ISSUE TITLE
New Feature Request: Audio Offset Setting

### ISSUE THREAD

**Issue by DaveTiff** [pkscout/script.audio.profiles#20]
Sorry I cant see a place to request a feature

Can the Audio Offset Value be included. I have a TV & Projector, I need a +200mS delay when using the projector, this would be a nice way to switch between.

Regards, Dave

**Comment by pkscout** [comment]
That is technically set per video (although you can set it as the default for all videos), and I don't see anyway to access or change that setting using the methods that Audio Profiles uses to create profiles.

**Comment by DaveTiff** [comment]
Ok, thanks

**Comment by STPKITT** [comment]
Hi,
I was about to propose the same feature and I'm glad I checked the closed issues first.
However, I just had a look at where that setting is stored and it's in the file /userdata/guisettings.xml in the following section:
`<defaultvideosettings>
<audiodelay>-0.075000</audiodelay>
</defaultvideosettings>`
So to me it looks to be exactly the same as other settings already controllable via this addon. Btw my niche usecase is that I need another delay between normal audio output and streaming it via Snapcast.
Would be great if this issue could be re-opened and tackled.
Thanks for reading anyway :-)

**Comment by pkscout** [comment]
Addons are not allowed to directly edit advanced settings. There is no JSON call or API to change it.

**Comment by STPKITT** [comment]
Oh, shame on me, I didn't know that!
Looking at https://kodi.wiki/view/Action_IDs though, Kodi offers Actions ID for audio delay, one of those being called "audiodelay" which I think when manually triggered by user input asks for a number to type in the desired delay. Isn't that how this addon could set the delay? According to https://forum.kodi.tv/showthread.php?tid=200079 it should work via Input.ExecuteAction. I'm no programmer though, this is just what I could find. What do you think?

**Comment by pkscout** [comment]
I understand your very strong desire for this, and I appreciate you looking at all these options, but please understand that I have been doing Kodi addons for over a decade and being working with this addon for nearly half that.  What you found lets you change the audio delay up or down, but not set it to a specific number.  Never mind having some way to get the current delay setting so it can be saved.  It's not going to work.  This request is closed for a reason, and I won't reply to any more comments on this.

**Comment by STPKITT** [comment]
Hi, I was reading about this topic again in order to find a workaround when I accidentally came across this:
https://forum.kodi.tv/showthread.php?tid=366795&pid=3123807#pid3123807 .

That refers to this: https://github.com/xbmc/xbmc/pull/22109 and is reflected in this Kodi Wiki entry: https://kodi.wiki/view/JSON-RPC_API/v13 .

So unless I'm getting it totally wrong (very sorry if that's the case) I guess we so far overlocked that change to Kodi which means this feature request isn't impossible to fulfil after all for an up-to-date Kodi version.

**Comment by pkscout** [comment]
Good catch.  I'll reopen this feature request.  It's not clear when I'm going to have time to look at this, but it will at least be on the list.

**Comment by STPKITT** [comment]
Great! No problem, it's just a hobby after all :-)
Greets!
